pkg/logs/input/listener/listener.go

```diff
@@ -10,19 +10,22 @@ import (
 
 	"github.com/DataDog/datadog-agent/pkg/logs/config"
 	"github.com/DataDog/datadog-agent/pkg/logs/pipeline"
+	"github.com/DataDog/datadog-agent/pkg/logs/restart"
 )
 
-// A Listener summons different protocol specific listeners based on configuration
+// Listener summons different protocol specific listeners based on configuration
 type Listener struct {
-	pp      pipeline.Provider
-	sources []*config.LogSource
+	pp        pipeline.Provider
+	sources   []*config.LogSource
+	listeners []restart.Stopper
```

Comment:
why not `[]*restart.Stopper`?

Reply:
Stopper is an interface, the concrete type can a pointer or a value